atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBackupIntegrationTest.java

```diff
@@ -0,0 +1,233 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfigManager;
+import com.palantir.atlasdb.containers.CassandraContainer;
+import com.palantir.atlasdb.containers.Containers;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.common.base.Throwables;
+import com.palantir.timestamp.TimestampBoundStore;
+
+public class CassandraTimestampBackupIntegrationTest {
+    private static final long INITIAL_VALUE = CassandraTimestampUtils.INITIAL_VALUE;
+    private static final long TIMESTAMP_1 = INITIAL_VALUE + 1000;
+    private static final long TIMESTAMP_2 = TIMESTAMP_1 + 1000;
+    private static final long TIMESTAMP_3 = TIMESTAMP_2 + 1000;
+
+    private static final long TIMEOUT_SECONDS = 5L;
+    private static final int POOL_SIZE = 32;
+
+    @ClassRule
+    public static final Containers CONTAINERS = new Containers(CassandraTimestampIntegrationTest.class)
+            .with(new CassandraContainer());
+
+    private final CassandraKeyValueService kv = CassandraKeyValueService.create(
+            CassandraKeyValueServiceConfigManager.createSimpleManager(CassandraContainer.KVS_CONFIG),
+            CassandraContainer.LEADER_CONFIG);
+    private final ExecutorService executorService = Executors.newFixedThreadPool(POOL_SIZE);
+    private final TimestampBoundStore timestampBoundStore = CassandraTimestampBoundStore.create(kv);
+    private final CassandraTimestampBackupRunner backupRunner = new CassandraTimestampBackupRunner(kv);
+
+    @Before
+    public void setUp() {
+        kv.dropTable(AtlasDbConstants.TIMESTAMP_TABLE);
+        backupRunner.createTimestampTable();
+    }
+
+    @After
+    public void close() {
+        kv.close();
+    }
+
+    @Test
+    public void canBackupWithDefaultValue() {
+        assertThat(backupRunner.backupExistingTimestamp(CassandraTimestampUtils.INITIAL_VALUE))
+                .isEqualTo(CassandraTimestampUtils.INITIAL_VALUE);
+    }
+
+    @Test
+    public void canBackupAlreadyStoredBound() {
+        timestampBoundStore.getUpperLimit();
+        timestampBoundStore.storeUpperLimit(TIMESTAMP_1);
+        assertThat(backupRunner.backupExistingTimestamp(TIMESTAMP_2)).isEqualTo(TIMESTAMP_1);
```

Comment:
This is successful - the argument to backup is a fall-back in the default case.

Actually, I realise that since I moved the `INITIAL_VALUE` to be a constant in `CassandraTimestampUtils`, we can instead always assume that that's the bound to be stored if no bound already exists.

(The reason why we need this whole backup idea is because I want `backup(); restore() TimestampBoundStore.get();` to have the same behaviour as just `TimestampBoundStore.get();`)